pkg/telemetry/cass_prometheus_resourcer.go

```diff
@@ -0,0 +1,459 @@
+      sourceLabels:
+      - mcac
+      targetLabel: __name__
+    - regex: org\.apache\.cassandra\.metrics\.hints_service\.hints_delays\-(\w+)
+      replacement: ${1}
+      sourceLabels:
+      - mcac
+      targetLabel: peer_ip
+    - regex: org\.apache\.cassandra\.metrics\.hints_service\.hints_delays\-(\w+)
+      replacement: mcac_hints_hints_delays
+      sourceLabels:
+      - mcac
+      targetLabel: __name__
+    - regex: org\.apache\.cassandra\.metrics\.hints_service\.([^\-]+)
+      replacement: mcac_hints_${1}
+      sourceLabels:
+      - mcac
+      targetLabel: __name__
+    - regex: org\.apache\.cassandra\.metrics\.memtable_pool\.(\w+)
+      replacement: mcac_memtable_pool_${1}
+      sourceLabels:
+      - mcac
+      targetLabel: __name__
+    - regex: com\.datastax\.bdp\.type\.performance_objects\.name\.cql_slow_log\.metrics\.queries_latency
+      replacement: mcac_cql_slow_log_query_latency
+      sourceLabels:
+      - mcac
+      targetLabel: __name__
+    - regex: org\.apache\.cassandra\.metrics\.read_coordination\.(.*)
+      replacement: $1
+      sourceLabels:
+      - mcac
+      targetLabel: read_type
+    - regex: jvm\.gc\.(\w+)\.(\w+)
+      replacement: ${1}
+      sourceLabels:
+      - mcac
+      targetLabel: collector_type
+    - regex: jvm\.gc\.(\w+)\.(\w+)
+      replacement: mcac_jvm_gc_${2}
+      sourceLabels:
+      - mcac
+      targetLabel: __name__
+    - regex: jvm\.memory\.(\w+)\.(\w+)
+      replacement: ${1}
+      sourceLabels:
+      - mcac
+      targetLabel: memory_type
+    - regex: jvm\.memory\.(\w+)\.(\w+)
+      replacement: mcac_jvm_memory_${2}
+      sourceLabels:
+      - mcac
+      targetLabel: __name__
+    - regex: jvm\.memory\.pools\.(\w+)\.(\w+)
+      replacement: ${2}
+      sourceLabels:
+      - mcac
+      targetLabel: pool_name
+    - regex: jvm\.memory\.pools\.(\w+)\.(\w+)
+      replacement: mcac_jvm_memory_pool_${2}
+      sourceLabels:
+      - mcac
+      targetLabel: __name__
+    - regex: jvm\.fd\.usage
+      replacement: mcac_jvm_fd_usage
+      sourceLabels:
+      - mcac
+      targetLabel: __name__
+    - regex: jvm\.buffers\.(\w+)\.(\w+)
+      replacement: ${1}
+      sourceLabels:
+      - mcac
+      targetLabel: buffer_type
+    - regex: jvm\.buffers\.(\w+)\.(\w+)
+      replacement: mcac_jvm_buffer_${2}
+      sourceLabels:
+      - mcac
+      targetLabel: __name__
+    - regex: (mcac_.*);.*(_micros_bucket|_bucket|_micros_count_total|_count_total|_total|_micros_sum|_sum|_stddev).*
+      replacement: ${1}${2}
+      separator: ;
+      sourceLabels:
+      - __name__
+      - prom_name
+      targetLabel: __name__
+    - action: labeldrop
+      regex: prom_name`
+
+// mustLabels() returns the set of labels essential to managing the Prometheus resources. These should not be overwritten by the user.
+func (cfg CassPrometheusResourcer) mustLabels() map[string]string {
+	return map[string]string{
+		"app.kubernetes.io/managed-by": "k8ssandra-operator",
```

Comment:
We have constants for some of these labels in `apis/k8ssandra/v1alpha1/constants.go`. Also have a look at `pkg/utils/labels.go`.

Reply:
Good idea, done.